src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/output/BigQuerySQLEngineOutputFormat.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine.output;
+
+import com.google.cloud.hadoop.io.bigquery.BigQueryFactory;
+import com.google.cloud.hadoop.io.bigquery.output.BigQueryOutputConfiguration;
+import com.google.cloud.hadoop.io.bigquery.output.ForwardingBigQueryFileOutputFormat;
+import io.cdap.plugin.gcp.bigquery.sink.BigQueryOutputFormat;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.FileAlreadyExistsException;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+/**
+ * An Output Format which extends {@link BigQueryOutputFormat}.
+ *
+ * This implementation overrides the implementation of
+ * {@link ForwardingBigQueryFileOutputFormat#checkOutputSpecs(JobContext)} in order to skip a check that prevents
+ * compression from being used.
```

Comment:
do we know why this was prevented to begin with? Seems odd that it was explicitly checking and preventing compression.

Reply:
The only hint in the docs is that, since not all input formats support the same compression codecs, they'd rather not to support compression at all: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/eaaebef7d5efaf51d97d37dfcf9c93a28e1d71ae/bigquery/src/main/java/com/google/cloud/hadoop/io/bigquery/output/ForwardingBigQueryFileOutputFormat.java#L75